Refactor/user and space popup

rdenarie opened the pull request.

rdenarie committed:
Revert "Task-48242: Make sure to display disabled and external users labels in user profile popup.  (#985)"

This reverts commit f13101b80828e7f6003536a725d39c2789737615.

rdenarie committed:
Refactor UserPopup Labels

boubaker commented on webapp/portlet/src/main/webapp/vue-apps/who-is-online-app/components/ExoWhoIsOnline.vue:
```diff
@@ -42,10 +43,16 @@ export default {
-      this.retrieveOnlineUsers();
-    }.bind(this), this.delay);
+    const component = this;
+    document.onreadystatechange = function () {
```
Use lambda function to reference `this` object, see https://community.exoplatform.com/portal/dw/activity?id=364805

rdenarie replied:
done

boubaker commented on webapp/portlet/src/main/webapp/vue-apps/who-is-online-app/components/ExoWhoIsOnline.vue:
```diff
@@ -42,10 +43,16 @@ export default {
+    const component = this;
+    document.onreadystatechange = function () {
+      if (document.readyState === 'complete' && !component.loaded) {
+        component.initOnlineUsers(component.$root.onlineUsers && component.$root.onlineUsers.users || []);
```
We may have to use this function outside the event listener to initialize users `href` data correctly.
You can put here the code of `this.initPopup()` instead of `this.initOnlineUsers()`

**Note**: please make sure the following instruction at the end of  `initOnlineUsers` function :
```javascript
      return this.$nextTick()
        .then(() => this.$root.$applicationLoaded());
```

rdenarie replied:
done

boubaker commented on translations.properties:
```diff
@@ -47,6 +47,7 @@ webapp/ProfileHeader.properties=webapp/portlet/src/main/resources/locale/portlet
 webapp/ProfileWorkExperience.properties=webapp/portlet/src/main/resources/locale/portlet/social/ProfileWorkExperience_en.properties
 webapp/SpaceAdministrationPortlet.properties=webapp/portlet/src/main/resources/locale/portlet/social/SpacesAdministrationPortlet_en.properties
 webapp/SpacesListApplication.properties=webapp/portlet/src/main/resources/locale/portlet/social/SpacesListApplication_en.properties
+webapp/UserPopup.properties=webapp/portlet/src/main/resources/locale/portlet/social/UserPopup_en.properties
```
IMO we should do the same for the spacePopover

rdenarie replied:
next step. My PR is still draft.
The spacePopup will be also in this PR

rdenarie committed:
Improve WhoIsOnline

rdenarie committed:
Refactor SpacePopup Labels

Before this fix, space popup labels are duplicated in a lot of properties files
This fix modify everywhere spacePopup is called to not use duplicated labels.
Labels which are no more necessary are removed

rdenarie committed:
Merge branch 'develop' into refactor/user-and-space-popup
